src/tf_service/client.py

```diff
@@ -42,47 +60,101 @@ def __init__(self, server_node_name, keepalive_period=None):
             if it dropped
         """
         tf2_ros.BufferInterface.__init__(self)
-        # All actual work is done by the C++ binding.
-        client_binding.roscpp_init_once()
-        self.client = client_binding.BufferClientBinding(server_node_name)
-        rospy.on_shutdown(client_binding.roscpp_shutdown)
-        # Keepalive timer must be implemented here. It would require a spinner
-        # in the C++ binding.
+
+        self._mutex = threading.Lock()
+        self._reconnection_mutex = threading.Lock()
+
+        can_transform_service_full: str = os.path.join(
+            server_node_name, CAN_TRANSFORM_SERVICE_NAME
+        )
+        lookup_transform_service_full: str = os.path.join(
+            server_node_name, LOOKUP_TRANSFORM_SERVICE_NAME
+        )
+
+        with self._reconnection_mutex:
+            self._can_transform_client = PersistentService(
+                can_transform_service_full, CanTransform
+            )
+            self._lookup_transform_client = PersistentService(
+                lookup_transform_service_full, LookupTransform
+            )
+
         if keepalive_period is not None:
-            self._keepalive_timer = rospy.Timer(keepalive_period,
-                                                self._keepalive_callback,
-                                                reset=True)
+            self._keepalive_timer = rospy.Timer(
+                keepalive_period, self._keepalive_callback, reset=True
+            )
 
-    @translate_exceptions
-    def wait_for_server(self, timeout=rospy.Duration(-1)):
+    def wait_for_server(self, timeout: Optional[rospy.Duration] = None) -> bool:
```

Comment:
Note: `rospy.Duration(-1)` only worked with the bindings, rospy needs `None` for the same effect.